src/GoodsList.jsx

```diff
@@ -0,0 +1,107 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+export class GoodsList extends React.Component {
+  state = {
+    isVisible: false,
+    isReversed: false,
+    sortBy: '',
+  }
+
+  showList = () => {
+    this.setState({
+      isVisible: true,
+    });
+  }
+
+  reverseList = () => {
+    this.setState(state => ({
+      isReversed: !state.isReversed,
+    }));
+  }
+
+  sortListByAlph = () => {
+    this.setState({
+      sortBy: 'alph',
+    });
+  }
+
+  sortListByLength = () => {
+    this.setState({
+      sortBy: 'length',

```

Comment:
```suggestion
      isReversed: false,
      sortBy: 'length',

```
Если нажать Sort by length потом reverse. Sort by length перестанет сортировать.